### PR TITLE
Bluetooth discovery : Countdown Timer and Snackbars

### PR DESCRIPTION
**Pull request by asaikarthikeya**
Fixes: #751 #596 #750 #742 

https://user-images.githubusercontent.com/70392921/123795470-acaac700-d901-11eb-870f-9cc3c82831c7.mp4

### CODE REVIEW

**Comment by asaikarthikeya** [comment]
@SebaDro Please review these changes. Thank you :)

**Review by SebaDro — CHANGES_REQUESTED** [review]
We should disable clicking the redisocver button when searching for new devices is an ongoing process. Otherwise, each time you click the rediscover button, additional 15 seconds will be added to the timer. E.g. if you click twice, discovery will last 30 seconds.

Maybe, during discovery, you can place the circular progress bar at the same position of the rediscover button and only show it for discovery time while hiding the circular progress bar. That would also fix the problem, that both icons are not aligned vertically in the middle.

**Comment by asaikarthikeya** [comment]
> We should disable clicking the redisocver button when searching for new devices is an ongoing process. Otherwise, each time you click the rediscover button, additional 15 seconds will be added to the timer. E.g. if you click twice, discovery will last 30 seconds.
> 
> Maybe, during discovery, you can place the circular progress bar at the same position of the rediscover button and only show it for discovery time while hiding the circular progress bar. That would also fix the problem, that both icons are not aligned vertically in the middle.

Nice Point Sir ! I will work on this and update it accordingly.

**Comment by asaikarthikeya** [comment]
@SebaDro Sir, I have removed the rescan option while discovery is in progress and Also aligned the views as you have proposed in the meeting ;-)

File: org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java [hunk 1]
```diff
@@ -343,7 +345,22 @@ public void onComplete() {
...
+                            public void onFinish() {
+                                // If discovering of a device takes more than 15 sec ,then show user discovery is finished
+                                // if the needed device is not found , rediscover button can be used.
+                                if (mBluetoothHandler.isBluetoothEnabled()) {
+                                    mProgressBar.setVisibility(View.GONE);
+                                    mRescanImageView.setVisibility(View.VISIBLE);
+                                    showSnackbar("Discovery Finished!");
+                                }
+                            }
+                        }.start();
```
Comment (SebaDro) [inline org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java:363]: Placing this into `onComplete()` does not has the desired effect. The `CountDownTimer` will only start after discovery has finished. So, why should we wait another 15 seconds for discovery? Wouldn't it be a better solution, to start the timer directly after clicking on the refresh button?

**Review by SebaDro — CHANGES_REQUESTED** [review]

File: org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java (hunk 1 above)
Comment (asaikarthikeya) [reply org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java]: Yes. I have placed this in `onStart()` and tested the functionality. Works fine in my device. Thank you testing this feature closely :)

**Review by SebaDro — CHANGES_REQUESTED** [review]
Great, starting the timer in `onStart()` is a much better solution.

Unfortunately, your implementation is a little bit too simple. The setting of 15 seconds is an arbitrary choice in the hope, it will match the duration od bluetooth discovery. You can't be sure that bluetooth discovery is finished after 15 only to show the "Discovery finished" snackbar. In addition, if Bluetooth discovery ends before the 15 seconds, the circular progress bar is still visible. You can simply check it by debugging the code. 

Therefore, I suggest two essential adjustments:

1. Stop Bluetooth discovery in the `onFinish()` event listener of the `CountTimer`.
2. Move changing visibility of the circular progressbar and refresh image as well as showing the snackbar to `onComplete()` of the Bluetooth handler.

In addition, you should use a String ressource for the snackbar text.

**Review by SebaDro — APPROVED** [review]

**Pull request merged**